Sources/ZcashLightClientKit/Block/Scan/BlockScanner.swift

```diff
@@ -56,7 +57,9 @@ extension BlockScannerImpl: BlockScanner {
             let scanSummary: ScanSummary
             let scanStartTime = Date()
             do {
-                scanSummary = try await self.rustBackend.scanBlocks(fromHeight: Int32(startHeight), limit: batchSize)
+                let fromState = try await service.getTreeState(BlockID(height: startHeight))
```

Comment:
`getTreeState` returns the tree state at the end of a block, and `fromState` needs to be as of the start of the first scanned block:
```suggestion
                let fromState = try await service.getTreeState(BlockID(height: startHeight - 1))
```